test/test_broker/iaas.CloudProviderClient.js

```diff
@@ -13,6 +13,191 @@ describe('iaas', function () {
       createClientSpy.reset();
     });
 
+    describe('#createDiskFromSnapshot', function () {
+      const diskId = 'sample-disk';
+      const snapshotId = 'sample-snapshot';
+      const volType = 'gp2';
+      const zone = 'zone';
+      const response = {
+        Volumes: [{
+          VolumeId: diskId,
+          Size: '4',
+          AvailabilityZone: 'zone',
+          VolumeType: 'type'
+        }]
+      };
+      it('should create disk from snapshot for aws', function () {
+        const client = new CloudProviderClient({
+          name: 'aws',
+          key: 'key',
+          keyId: 'keyId',
+          region: 'region'
+        });
+        const createVolStub = sinon.stub(client.blockstorage, 'createVolume');
+        createVolStub.withArgs({
+          AvailabilityZone: zone,
+          SnapshotId: snapshotId,
+          VolumeType: volType,
+          TagSpecifications: [{
+            ResourceType: 'volume',
+            Tags: [{
+              Key: 'customkey',
+              Value: 'customvalue'
+            }]
+          }]
+        }).returns({
+          promise: () => Promise.resolve(response)
+        });
+        const waitForStub = sinon.stub(client.blockstorage, 'waitFor');
+        waitForStub.withArgs('volumeAvailable', {
+            VolumeIds: [diskId]
+          })
+          .returns({
+            promise: () => Promise.resolve(response)
+          });
+        return client
+          .createDiskFromSnapshot(snapshotId, zone, {
+            volumeType: 'gp2',
+            tags: {
+              customkey: 'customvalue'
+            }
+          })
+          .then(disk => {
+            expect(disk.volumeId).to.eql(diskId);
+            expect(disk.size).to.eql('4');
+            expect(disk.zone).to.eql('zone');
+            expect(disk.type).to.eql('type');
```

Comment:
Same as above, please verify all keys in response.

Reply:
Done in latest commit